cli/tests/pcluster/config/test_update_policy.py

```diff
@@ -453,3 +453,357 @@ def test_compute_fleet_stop_fail_reason_and_actions_needed(
     assert_that(UpdatePolicy.COMPUTE_FLEET_STOP.action_needed(change_mock, patch_mock)).is_equal_to(
         expected_actions_needed
     )
+
+
+@pytest.mark.parametrize(
+    "is_fleet_stopped, key, path, old_value, new_value, update_strategy, expected_result, expected_fail_reason, "
+    "expected_actions_needed, scheduler",
+    [
+        pytest.param(
+            True,
+            "SharedStorage",
+            [],
+            None,
+            {"MountDir": "/ebs3", "Name": "ebs3", "StorageType": "Ebs", "EbsSettings": {"VolumeType": "gp3"}},
+            None,
+            True,
+            None,
+            None,
+            "slurm",
+            id="stopped fleet and add new EBS section",
+        ),
+        pytest.param(
+            True,
+            "SharedStorage",
+            [],
+            {"MountDir": "/ebs3", "Name": "ebs3", "StorageType": "Ebs", "EbsSettings": {"VolumeType": "gp3"}},
+            None,
+            None,
+            True,
+            "All compute nodes must be stopped",
+            "Stop the compute fleet with the pcluster update-compute-fleet command",
+            "slurm",
+            id="stopped fleet and remove EBS section",
+        ),
+        pytest.param(
+            False,
+            "SharedStorage",
+            [],
+            None,
+            {"MountDir": "/ebs3", "Name": "ebs3", "StorageType": "Ebs", "EbsSettings": {"VolumeType": "gp3"}},
+            None,
+            False,
+            "All compute nodes must be stopped or QueueUpdateStrategy must be set",
+            "Stop the compute fleet with the pcluster update-compute-fleet command, or set QueueUpdateStrategy in the "
+            "configuration used for the 'update-cluster' operation",
+            "slurm",
+            id="running fleet and EBS changed with no update strategy set",
+        ),
+        pytest.param(
+            False,
+            "SharedStorage",
+            [],
+            None,
+            {"MountDir": "/ebs3", "Name": "ebs3", "StorageType": "Ebs", "EbsSettings": {"VolumeType": "gp3"}},
+            None,
+            None,
+            "Update actions are not currently supported for the 'SharedStorage' parameter",
+            "Restore the parameter 'SharedStorage'. If you need this change, please consider creating a new cluster "
+            "instead of updating the existing one.",
+            "awsbatch",
+            id="running fleet and EBS changed with no update strategy set with batch scheduler",
```

Comment:
Typically we refer to the AWS Batch scheduler with the name "awsbatch" rather than "batch".

Reply:
Changed the id